django/template/loaders/base.py

```diff
@@ -44,3 +93,10 @@ def reset(self):
         templates or cached loader modules).
         """
         pass
+
+    @property
+    def supports_recursion(self):
+        """
+        Used during deprecation period of non-recursive loaders.
```

Comment:
I'm fine with whatever you think is best. I think the number of people writing custom template loaders is low, so docs might be sufficient if a warning will be more effort.